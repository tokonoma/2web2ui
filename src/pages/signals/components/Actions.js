import React from 'react';
import PropTypes from 'prop-types';
import { Warning, CheckCircleOutline } from '@sparkpost/matchbox-icons';
import { formatDate } from 'src/helpers/date';
import useHibanaOverride from 'src/hooks/useHibanaOverride';
import { Heading } from 'src/components/text';
import { ExternalLink, PageLink } from 'src/components/links';
import Callout from 'src/components/callout';
import OGStyles from './Actions.module.scss';
import hibanaStyles from './ActionsHibana.module.scss';

const Action = ({ content, link, type = 'bad', internal = false, linkText = 'Learn More' }) => {
  const styles = useHibanaOverride(OGStyles, hibanaStyles);

  let iconMarkup;

  const linkMarkup =
    link &&
    (internal ? (
      <PageLink to={link} className={styles.Link}>
        {linkText}
      </PageLink>
    ) : (
      <ExternalLink to={link} className={styles.Link}>
        {linkText}
      </ExternalLink>
    ));

  if (type === 'bad') {
    iconMarkup = (
      <div className={styles.IconBad}>
        <Warning size={25} />
      </div>
    );
  }

  if (type === 'good') {
    iconMarkup = (
      <div className={styles.IconGood}>
        <CheckCircleOutline size={25} />
      </div>
    );
  }

  if (type === 'warning') {
    iconMarkup = (
      <div className={styles.IconWarning}>
        <Warning size={25} />
      </div>
    );
  }

  return (
    <div className={styles.ActionWrapper}>
      <div className={styles.Action}>
        {iconMarkup}
        <p>
          {content} {linkMarkup}
        </p>
      </div>
      <hr className={styles.Dash} />
    </div>
  );
};

const Actions = ({ actions, date, empty }) => {
  const styles = useHibanaOverride(OGStyles, hibanaStyles);

  return (
    <div className={styles.Wrapper}>
      <div className={styles.Title}>
        <Heading as="h6" className={styles.TitleText}>
          Recommendations
          {date && ` â€“ ${formatDate(date)}`}
        </Heading>
      </div>
      {!empty && actions.map((props, i) => <Action key={i} {...props} />)}
      {empty && <Callout height="100px">No actions to display at this time.</Callout>}
    </div>
  );
};

Actions.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.oneOf(['good', 'bad', 'warning']),
      content: PropTypes.node,
      link: PropTypes.string,
    }),
  ),
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
};

export default Actions;
