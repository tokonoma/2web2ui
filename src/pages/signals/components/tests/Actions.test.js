import { shallow } from 'enzyme';
import React from 'react';
import Actions from '../Actions';

jest.mock('src/hooks/useHibanaOverride', () => state => state);

describe('Signals Recommended Actions Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      actions: [
        { content: 'foo', link: 'http://google.com' },
        { content: 'bar', type: 'good' },
        { content: 'baz', type: 'warning' },
        {
          content: 'lorem',
          type: 'bad',
          link: '/internal',
          internal: true,
          linkText: 'View Internal Link',
        },
      ],
      date: new Date('2018-01-01'),
      empty: false,
    };
    wrapper = shallow(<Actions {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.find('Action').forEach(node => {
      expect(node.dive()).toMatchSnapshot();
    });
  });

  it('renders empty state when empty', () => {
    wrapper.setProps({ empty: true });
    expect(wrapper).toMatchSnapshot();
  });
});
