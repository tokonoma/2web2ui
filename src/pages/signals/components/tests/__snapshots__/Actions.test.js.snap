// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals Recommended Actions Component renders correctly 1`] = `
<div
  className="Wrapper"
>
  <div
    className="Title"
  >
    <Heading
      as="h6"
      className="TitleText"
    >
      Recommendations
       – Dec 31 2017
    </Heading>
  </div>
  <Action
    content="foo"
    key="0"
    link="http://google.com"
  />
  <Action
    content="bar"
    key="1"
    type="good"
  />
  <Action
    content="baz"
    key="2"
    type="warning"
  />
  <Action
    content="lorem"
    internal={true}
    key="3"
    link="/internal"
    linkText="View Internal Link"
    type="bad"
  />
</div>
`;

exports[`Signals Recommended Actions Component renders correctly 2`] = `
<div
  className="ActionWrapper"
>
  <div
    className="Action"
  >
    <div
      className="IconBad"
    >
      <Warning
        size={25}
      />
    </div>
    <p>
      foo
       
      <ExternalLink
        className="Link"
        to="http://google.com"
      >
        Learn More
      </ExternalLink>
    </p>
  </div>
  <hr
    className="Dash"
  />
</div>
`;

exports[`Signals Recommended Actions Component renders correctly 3`] = `
<div
  className="ActionWrapper"
>
  <div
    className="Action"
  >
    <div
      className="IconGood"
    >
      <CheckCircleOutline
        size={25}
      />
    </div>
    <p>
      bar
       
    </p>
  </div>
  <hr
    className="Dash"
  />
</div>
`;

exports[`Signals Recommended Actions Component renders correctly 4`] = `
<div
  className="ActionWrapper"
>
  <div
    className="Action"
  >
    <div
      className="IconWarning"
    >
      <Warning
        size={25}
      />
    </div>
    <p>
      baz
       
    </p>
  </div>
  <hr
    className="Dash"
  />
</div>
`;

exports[`Signals Recommended Actions Component renders correctly 5`] = `
<div
  className="ActionWrapper"
>
  <div
    className="Action"
  >
    <div
      className="IconBad"
    >
      <Warning
        size={25}
      />
    </div>
    <p>
      lorem
       
      <PageLink
        className="Link"
        to="/internal"
      >
        View Internal Link
      </PageLink>
    </p>
  </div>
  <hr
    className="Dash"
  />
</div>
`;

exports[`Signals Recommended Actions Component renders empty state when empty 1`] = `
<div
  className="Wrapper"
>
  <div
    className="Title"
  >
    <Heading
      as="h6"
      className="TitleText"
    >
      Recommendations
       – Dec 31 2017
    </Heading>
  </div>
  <Callout
    height="100px"
  >
    No actions to display at this time.
  </Callout>
</div>
`;
